### DISCUSSION TITLE
Scripts and parts

### DISCUSSION THREAD

**Discussion by pierremilcent** [5axes/Calibration-Shapes#8] [Q&A]
Hello,
i have added the plugin to Cura 4.8. I want to do a retract tower test. How to use your script in conjonction with the part ? I clicked the "copy script" from "Parts for calibration" menu, but don't know how to change the retract value at each step.
How the script is corelated to the part and how to activate it ? I guess the script is changing retract value at some defined height ?
Thanks

**Comment by pierremilcent** [comment]
I found that after Cura restart when i activated "copy script" new scripts are available in the Post processing menu. And through the "RetractTower" script i can now change retract values.
Nice !! thanks to myself.

**Reply by 5axes** [reply]
Thank you for your comment. I will add in the official release a small message when you copy the scripts to inform the user to restart Cura !

**Reply by pierremilcent** [reply]
I just printed a PLA temp tower with the scrpit post treatment active but the temperature changed only between the 4 first lines at 205°, my default temp, and then all upper lines are at 220° with not decrease.
![2021-01-28 19_54_33-Greenshot](https://user-images.githubusercontent.com/38214885/106185229-d4ef9780-61a2-11eb-94be-fe10e8ad860a.png)
![2021-01-28 19_53_08-Sans titre - Ultimaker Cura 4 8 0](https://user-images.githubusercontent.com/38214885/106185233-d620c480-61a2-11eb-9736-8607ba5a8ad0.png)

**Reply by 5axes** [reply]
That's normal you should not activate the fan option. For each temperature fan speed will change to 100%  40% and 0% and in this case the temp geometry  is to small.

**Reply by pierremilcent** [reply]
Thanks i will try tomorrow. Should be advertised also in your message with the scripts.

**Reply by 5axes** [reply]
Not so easy , so many possibilities. The goal of this plugin for the moment is just to give some tools, not a working method ready to run. Perhap's a futur enhancement

**Reply by pierremilcent** [reply]
OK the test with your recommendation is working fine for me : from 220 to 180° decreasing 5° each 38 layers starting at the 5th. i disabled fan option as you said. 
Thanks.

**Reply by 5axes** [reply]
I will try to put more information available  in the next days on this repository [wiki ](https://github.com/5axes/Calibration-Shapes/wiki)

**Reply by tonnic** [reply]
> erature fan speed will

So how do we use the fan tower??   If it isn't intended for the temp tower, how is it supposed to be used?

**Reply by 5axes** [reply]
You can use it with a temptower. But it's mainly used with PETG, with PLA fan speed is set to 100% most of the times. To use it try to find the best temperatur for your filament. And then in the field you can define the fan speed sequence Ie: 40,35,30,25,20,15,10,5,0

**Reply by tonnic** [reply]
oh ok.. that's what I thought, but for some reason I thought I read that it isn't supposed to be used with the temptower because it is too small or something...

**Reply by tonnic** [reply]
> You can use it with a temptower. But it's mainly used with PETG, with PLA fan speed is set to 100% most of the times. To use it try to find the best temperatur for your filament. And then in the field you can define the fan speed sequence Ie: 40,35,30,25,20,15,10,5,0

So you would change the temp increment to zero and then only have the fan speed sequence?   It will change on the layer change as the temp would, but at this point only fan?

**Reply by 5axes** [reply]
Not necessary. in some case you would try to analyse the difference between 3 different temperatur and 3 Fan speed.  
![image](https://user-images.githubusercontent.com/11015345/115145956-b76cf400-a054-11eb-9362-b16a5d66afd9.png)

**Reply by pierremilcent** [reply]
This should be detailed in the wikipage ? shouldn't it ?

**Reply by 5axes** [reply]
Not sure that anyone is reading the wiki

**Reply by pierremilcent** [reply]
So why not to provide a link in the online help to it ?
If not i guess the plugin will be not totally used with its capabilities ...

**Reply by 5axes** [reply]
There is already a link to the wiki. But as this fan settings is not an usual procedure to test or calibrate settings I don't want to give to many "complex" information. For the moment I'm fitting against the message of people Who claim that this plugin is bullshit because they cannot print the tower with their 0.8 nozzle ! So the fan option is not a main problem after all    🤣

**Reply by pierremilcent** [reply]
this is fair ... good luck with them ... Merci.

**Reply by tonnic** [reply]
I also agree that this should be in the wiki so people can read up on your program features.  Regardless of other complaints, a complete wiki is always beneficial.
